### PR TITLE
Add Safello as purchase option for ETH

### PR DESCRIPTION
**Pull request by t4t5**
I noticed that there are currently no options in the MetaMask extension for Europeans who want to quickly exchange their fiat for ETH ([Wyre](https://dash.sendwyre.com/sign-up) only accepts US credit cards).

This PR adds the Safello Quickbuy widget (which accepts European credit cards) as an option for buying ether straight from MetaMask.

The widget supports the currencies EUR, GBP, SEK, NOK and DKK and will automatically choose the best payment method based on the user's country.

One perk with this integration is also that it doesn’t require redirecting to an external website – the whole purchase can be completed within the MetaMask window itself.

**Preview:**

metamask

Feedback is welcome! Let me know if you have any questions. 😊

### CODE REVIEW

**Comment by bdresser** [comment]
@t4t5 thanks for this.

Can you give us some info on the Yoti app and how many people use it? Is there a way to enable purchaes via Safello without requiring this app? 

We have plans to integrate Simplex shortly, which should solve the same usecase (ETH onboarding for non-US credit cards) without requiring a specific auth app. 

(cc @danjm)

**Comment by bdresser** [comment]
@t4t5 thanks for this.

Can you give us some info on the Yoti app and how many people use it? Is there a way to enable purchases via Safello without requiring this app? 

We have plans to integrate Simplex shortly, which should solve the same usecase (ETH onboarding for non-US credit cards) without requiring a specific auth app. 

(cc @danjm)

**Comment by ubuwaits** [comment]
Hi, @bdresser and @danjm. I also work at Safello, so I wanted to jump in and answer your questions.

We don't have specific numbers on Yoti users, but our data shows that the conversation rate for Yoti users (even when users must sign up while purchasing) is on-par or higher than other solutions in this space. That said, I understand the issue of having a separate app to download. We are working with Yoti on a new integration that can be completely integrated into the UI. 

For the time being, we still believe the Yoti app provides a good experience for customers. New users can have their identity confirmed in less than 5 minutes and repeat purchases just require a single tap in the app to approve. For customers in Sweden, Denmark and Norway, we use the standard national eID solutions that are very widely used and provide an even faster purchasing experience.

Happy to answer any other questions you might have.

**Comment by danfinlay** [comment]
> One perk with this integration is also that it doesn’t require redirecting to an external website – the whole purchase can be completed within the MetaMask window itself.

In our experience at MetaMask, the "action view" is actually not a great place to ask users to fill out forms: Every time they click to another view, the window will collapse, and all form fields will be wiped clear. It's a much better experience to direct to a full-page experience, and doing that also greatly reduces the footprint on the MetaMask application, allowing a much lighter-weight merge & link.

**Comment by t4t5** [comment]
Updated so that i redirects to an external page instead:

Screenshot 2019-07-10 at 13 46 24

**Comment by bdresser** [comment]
hey @t4t5 just wanted to bump my questions above:

Can you give us some info on the Yoti app and how many people use it? Is there a way to enable purchaes via Safello without requiring this app?

File: app/_locales/en/messages.json [hunk 1]
```diff
@@ -1194,6 +1200,9 @@
   "settings": {
     "message": "Settings"
   },
+  "safelloBuy": {
+    "message": "Buy with Safello"
```
Comment (Gudahtt) [inline app/_locales/en/messages.json]: Do we still need this message? It seems... overly burdensome on the translation team to ask for "Buy with Safello" _and_ "Buy ETH with Safello" when we could just use the former in both places.

**Review by Gudahtt — APPROVED** [review]
Looks good!

**Comment by t4t5** [comment]
> hey @t4t5 just wanted to bump my questions above:
> 
> Can you give us some info on the Yoti app and how many people use it? Is there a way to enable purchaes via Safello without requiring this app?

Yoti claims to have over 5 million customers. We can verify from the Google Play Store that the Android app has over a million installs.

We are open to other verification methods, but I think it's hard to find something that works as smoothly and reliably as Yoti in Europe at this time. 😊

**Review by miranda76 — APPROVED** [review]

**Comment by whymarrh** [comment]
Hey, @t4t5, thanks for proposing this—I believe Safello is in talks with our team (@JSON-LEE13) about some of the finer details around merging this. I'm going to close this for now, so that we can focus on other PRs, and we can re-open this when we're closer to merging it.

**Pull request closed without merging**